tools/autograd/gen_variable_factories.py

```diff
@@ -34,7 +34,8 @@ def gen_variable_factories(out, declarations, template_path):
     function_definitions = []
     for decl in declarations:
         has_tensor_options = any(a["simple_type"] == "TensorOptions" for a in decl["arguments"])
-        if has_tensor_options or decl["name"].endswith("_like"):
+        is_namespace_fn = 'namespace' in decl['method_of']
```

Comment:
Oh, that's interesting. What does `method_of` do again?

Reply:
This is saying we should only generate a variable factory corresponding to a tensor factory function if it is actually a namespace function, i.e. has variant: function.  The previous code would generate a variable factory if it had a TensorOptions parameter, but this had a bad effect when combined with the `variants: []` concept when I moved some of the Type-parameter functions to be TensorOptions-parameter functions (i.e. it would generate variable factories even though a corresponding factory function didn't exist).